sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -627,21 +627,22 @@ abstract class StreamExecution(
       inputPlan.schema,
       new CaseInsensitiveStringMap(options.asJava))
     val writeBuilder = table.newWriteBuilder(info)
-    outputMode match {
+    val write = outputMode match {
```

Comment:
This probably deserves a separate PR. Required distribution and sort for streaming plans is a separate discussion we should have once we cover the batch plans.

I could technically keep using the old deprecated method here and just adapt `InMemoryTable` implementation until we have a proper plan. As an alternative, I can move this piece to use `Write`s so that data sources that build `Write`s will work correctly as an intermediate step.

Reply:
I definitely prefer separate PRs for batch and streaming, since streaming tends to touch a lot more classes.

Reply:
Bringing this logic to streaming would require introducing new plans. There are a few old TODO items for this in the code. For now, I'll submit a separate PR for streaming to use the `Write` abstraction.

Reply:
I've submitted PR #31093.

Reply:
The PR was merged but I am not resolving this so that other reviewers can see this thread.